security/rbac/engine/engine_test.go

```diff
@@ -0,0 +1,264 @@
+	"strings"
+	"testing"
+
+	pb "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v2"
+	cel "github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types/ref"
+	interpreter "github.com/google/cel-go/interpreter"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type programMock struct {
+	out ref.Val
+	err error
+}
+
+func (mock programMock) Eval(vars interface{}) (ref.Val, *cel.EvalDetails, error) {
+	return mock.out, nil, mock.err
+}
+
+type valMock struct {
+	val interface{}
+}
+
+func (mock valMock) ConvertToNative(typeDesc reflect.Type) (interface{}, error) {
+	return nil, nil
+}
+
+func (mock valMock) ConvertToType(typeValue ref.Type) ref.Val {
+	return nil
+}
+
+func (mock valMock) Equal(other ref.Val) ref.Val {
+	return nil
+}
+
+func (mock valMock) Type() ref.Type {
+	return nil
+}
+
+func (mock valMock) Value() interface{} {
+	return mock.val
+}
+
+var (
+	emptyActivation     = interpreter.EmptyActivation()
+	unsuccessfulProgram = programMock{out: nil, err: status.Errorf(codes.InvalidArgument, "Unsuccessful program evaluation")}
+	errProgram          = programMock{out: valMock{"missing attributes"}, err: status.Errorf(codes.InvalidArgument, "Successful program evaluation to an error result -- missing attributes")}
+	trueProgram         = programMock{out: valMock{true}, err: nil}
+	falseProgram        = programMock{out: valMock{false}, err: nil}
+
+	allowMatchEngine = &policyEngine{action: pb.RBAC_ALLOW, programs: map[string]cel.Program{
+		"allow match policy1": unsuccessfulProgram,
+		"allow match policy2": trueProgram,
+		"allow match policy3": falseProgram,
+		"allow match policy4": errProgram,
+	}}
+	denyFailEngine = &policyEngine{action: pb.RBAC_DENY, programs: map[string]cel.Program{
+		"deny fail policy1": falseProgram,
+		"deny fail policy2": falseProgram,
+		"deny fail policy3": falseProgram,
+	}}
+	denyUnknownEngine = &policyEngine{action: pb.RBAC_DENY, programs: map[string]cel.Program{
+		"deny unknown policy1": falseProgram,
+		"deny unknown policy2": unsuccessfulProgram,
+		"deny unknown policy3": errProgram,
+		"deny unknown policy4": falseProgram,
+	}}
+)
+
+func TestNewAuthorizationEngine(t *testing.T) {
+	tests := map[string]struct {
+		input   []*pb.RBAC
+		wantErr string
+		errStr  string
+	}{
+		"too few rbacs": {
+			input:   []*pb.RBAC{},
+			wantErr: "code = InvalidArgument desc = must provide 1 or 2 RBACs",
+			errStr:  "Expected wrong number of RBACs error for 0 RBACs",
+		},
+		"too many rbacs": {
+			input:   []*pb.RBAC{{}, {}, {}},
+			wantErr: "code = InvalidArgument desc = must provide 1 or 2 RBACs",
+			errStr:  "Expected wrong number of RBACs error for 3 RBACs",
+		},
+		"wrong rbac actions": {
+			input:   []*pb.RBAC{{Action: pb.RBAC_ALLOW}, {Action: pb.RBAC_DENY}},
+			wantErr: "code = InvalidArgument desc = when providing 2 RBACs, must have 1 DENY and 1 ALLOW in that order",
+			errStr:  "Expected wrong RBAC actions error for ALLOW followed by DENY",
+		},
```

Comment:
You also need a test for a good RBAC creation. Here it would be good to have one RBAC and two RBAC(s) cases.